lib/jsdom/living/helpers/selectors.js

```diff
@@ -1,47 +1,92 @@
 "use strict";
 
-const nwsapi = require("nwsapi");
-
+const domSelector = require("@asamuzakjp/dom-selector");
+const DOMException = require("../generated/DOMException");
 const idlUtils = require("../generated/utils");
 
-function initNwsapi(node) {
-  const { _globalObject, _ownerDocument } = node;
-
-  return nwsapi({
-    document: idlUtils.wrapperForImpl(_ownerDocument),
-    DOMException: _globalObject.DOMException
-  });
-}
-
-exports.matchesDontThrow = (elImpl, selector) => {
-  const document = elImpl._ownerDocument;
-
-  if (!document._nwsapiDontThrow) {
-    document._nwsapiDontThrow = initNwsapi(elImpl);
-    document._nwsapiDontThrow.configure({
-      LOGERRORS: false,
-      VERBOSITY: false,
-      IDS_DUPES: true,
-      MIXEDCASE: true
-    });
+exports.matchesDontThrow = (selectors, elementImpl) => {
+  let matched;
+  try {
+    const element = idlUtils.wrapperForImpl(elementImpl);
+    matched = domSelector.matches(selectors, element);
+  } catch (e) {
+    matched = false;
   }
+  return matched;
+};
 
-  return document._nwsapiDontThrow.match(selector, idlUtils.wrapperForImpl(elImpl));
+exports.matches = (selectors, elementImpl, globalObject) => {
+  let matched;
+  try {
```

Comment:
I noticed that the global `window` is accessible via `document.defaultView`.
So if dom-selector throws an exception like `throw new document.defaultView.DOMException()`, these try-catch can be completely removed, right?

Reply:
Yes, that's right!

Reply:
Thanks for the  quick reply.
I will rewrite the code and PR.